libs/time/parse.go

```diff
@@ -0,0 +1,20 @@
+package time
+
+import (
+	"fmt"
+	"time"
+)
+
+// ParseStringToTime takes a string pointer and returns a time value formatted as time.RFC3339.
+// The provided value must be parsable to time.RFC3339.
+// If the pointer value is nil then nil is returned
+func ParseStringToTime(value *string) (*time.Time, error) {
```

Comment:
it seems like everywhere else we use `time.Parse` directly [1]. this seems to add little value over that except checking the pointer first - but that only seems relevant to the use of this to parse valid to / valid from. checking [that usage](https://github.com/brave-intl/bat-go/blob/8b5a5e133090b9adfa011e06facab158074c620f/services/skus/datastore.go#L1566-L1576) more closely, it seems like the decision to not return an error in the case that the pointer is nil is incorrect. time limited credentials must have these defined AFAIK.

let's remove this and fix the above

[1]:
```
[user@work:~/source/bat-go]% rg ParseStringToTime
services/skus/datastore_test.go
395:    to, err := timeutils.ParseStringToTime(&vTo)
398:    from, err := timeutils.ParseStringToTime(&vFrom)
498:    validTo, err := timeutils.ParseStringToTime(&to)
502:    validFrom, err := timeutils.ParseStringToTime(&from)

services/skus/datastore.go
1566:                   validTo, err := timeutils.ParseStringToTime(so.ValidTo.Value())
1572:                   validFrom, err := timeutils.ParseStringToTime(so.ValidFrom.Value())
```